gobblin-api/src/main/java/gobblin/configuration/ImmutableWorkUnitState.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.configuration;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.util.Properties;
+
+import gobblin.source.extractor.Watermark;
+
+
+/**
+ * An immutable version of {@link WorkUnitState}.
+ *
+ * @author ynli
+ */
+public class ImmutableWorkUnitState extends WorkUnitState {
+
+  public ImmutableWorkUnitState(WorkUnitState workUnitState) {
+    super(workUnitState.getWorkunit());
```

Comment:
Not necessary since `WorkUnitState.getWorkUnit()` returns an `ImmutableWorkUnit`.